### ISSUE TITLE
Deadlock issue caused by timeout mutex locks

### ISSUE THREAD

**Issue by badrbouslikhin** [embassy-rs/teleprobe#6]
Hello,

I'm experiencing issues with the mutex locks, as it appears to be causing a deadlock. Here's the code snippet that triggers the problem:

https://github.com/embassy-rs/teleprobe/blob/6eb24c3182ef94665fe3a17fb10acd85a3c9cda2/teleprobe/src/server.rs#L196-L200

To resolve the issue, I made the following change:

```rust
let default_timeout = cx.lock().config.default_timeout;
let max_timeout = cx.lock().config.max_timeout;
let timeout_secs = args.timeout.unwrap_or(default_timeout).min(max_timeout);
```

This modification seems to fix the problem. Would you like me to open a pull request to implement this change?

**Comment by Dirbaio** [comment]
Yes please.

Strange that it works for me though. Huh.

**Comment by Dirbaio** [comment]
fixed in 0099ddaeacde7e5d9b0930a4c6e029e89c308982
